src/main/java/xyz/nucleoid/plasmid/storage/ServerStorage.java

```diff
@@ -0,0 +1,42 @@
+package xyz.nucleoid.plasmid.storage;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.util.Identifier;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implemented in classes that will store NBT data on the server.
+ * This interface is useful for storing data about games, e.g. player scores.
+ *
+ * @see {@link net.minecraft.world.PersistentState}
+ */
+public interface ServerStorage {
+    Map<Identifier, ServerStorage> STORAGES = new HashMap<>();
+
+    /**
+     * Adds a {@link ServerStorage} to the {@link #STORAGES} map and returns the {@link ServerStorage}.
+     * Use this to have your {@link ServerStorage} be saved and loaded.
+     * @param key - The id of the storage.
+     * @param storage - The {@link ServerStorage} to add to the {@link #STORAGES} map.
+     * @param <S> - The type of {@link ServerStorage}.
+     * @return - The supplied {@link ServerStorage}.
+     */
+    static <S extends ServerStorage> S createStorage(Identifier key, S storage) {
```

Comment:
Should this be allowed to be called after world load? I guess it should be fine under the circumstance that nothing has already been saved for that key, but it'd leave the storage to never be loaded

Reply:
Hm, yeah fair point. Perhaps it should throw an exception if it's called after the storages have been loaded?

Reply:
Yeah, that'd work 🙂

Reply:
Alternatively- although, this may be a little more annoying to use, we could force instances to be registered through the PersistentState object

Reply:
Yeah, fair.
But preferably this system should be very simple and easy to use so just throwing an exception works good in my opinion, and I don't imagine people trying to create new storages after it loads since it wouldn't load that storage at all and the exception works as a good notifier and catch for that.

Reply:
Yeah, thinking about it more, that's definitely the better solution :)